### PR TITLE
chore(ci): Loop mentions for reviewers, exclude autorelease PRs 

### PR DESCRIPTION
**Pull request by universal-itengineer**
## Description

Improve PR notificator bot:

- Add Loop-style names for reviewers.
- Exclude PRs by labels, e.g. PRs from release-please.

## Checklist
- [x] Changes were tested manually.

### CODE REVIEW

**Comment by universal-itengineer** [comment]
image

**Review by Isteb4k — APPROVED** [review]

File: .github/scripts/fetch_prs.mjs [hunk 1]
```diff
@@ -51,56 +56,126 @@ async function fetchReviewerDetails(login) {
...
-    throw error;
-  }
+  // const url = process.env.LOOP_WEBHOOK_URL;
+  // try {
+  //   await axios.post(url, { text: summary });
+  //   console.log('Summary sent successfully.');
+  // } catch (error) {
+  //   console.error('Error sending summary to Loop:', error);
+  //   throw error;
+  // }
```
Comment (diafour) [inline .github/scripts/fetch_prs.mjs:177]: Commented for debug?

File: .github/scripts/fetch_prs.mjs [hunk 2]
```diff
@@ -51,56 +56,126 @@ async function fetchReviewerDetails(login) {
...
+    for (const reviewer of pr.requested_reviewers) {
+      const details = await fetchReviewerDetails(reviewer.login);
+      const rewName = details.name ? details.name.replace(/ /g, '.').toLowerCase() : 'No name';
+      fetchedReviewers.push(details ? `${details.login} (@${rewName})` : reviewer.login);
+    }
+  }
+
+  // Add reviewers from reviews
+  const reviews = await fetchReviewsForPR(pr.number);
+  const uniqueReviewers = new Set(fetchedReviewers); // To avoid duplicates
```
Comment (diafour) [inline .github/scripts/fetch_prs.mjs:89]: You can use logins for this this set to reduce formatting dublicates: `${details.login} (@${rewName})`

File: .github/scripts/fetch_prs.mjs [hunk 3]
```diff
@@ -51,56 +56,126 @@ async function fetchReviewerDetails(login) {
...
   let summary = `## ${project} PRs ${now.format('YYYY-MM-DD')}\n\n`;
 
-  if (reviewRequired.length === 0) {
+  if (prs.length === 0) {
     summary += `:tada: No review required for today\n`;
     return summary;
   }
 
+  if (readyForMerge.length > 0) {
+    const readyForMergePRsInfo = await Promise.all(readyForMerge.map(pr => formatPR(pr, ' - **Ready for merge**')));
```
Comment (diafour) [inline .github/scripts/fetch_prs.mjs:152]: Second argument is not used (it is commented in formatPR), also, not needed for PRs in a separate section.

```suggestion
    const readyForMergePRsInfo = await Promise.all(readyForMerge.map(pr => formatPR(pr)));
```

File: .github/scripts/fetch_prs.mjs [hunk 4]
```diff
@@ -19,10 +19,11 @@ import moment from 'moment';
 
 const owner = 'deckhouse';
 const repo = 'virtualization';
-const project = 'DVP';
-const defaultLogin = 'Almighty PM';
+const project = ':dvp: DVP';
+const defaultLogin = '@anton.nikoonov';
```
Comment (diafour) [inline .github/scripts/fetch_prs.mjs:23]: nikioonov - typo?

Comment (universal-itengineer) [reply .github/scripts/fetch_prs.mjs]: Yes, fixed

File: .github/scripts/fetch_prs.mjs [hunk 5]
```diff
@@ -51,56 +56,126 @@ async function fetchReviewerDetails(login) {
...
-  const recent = reviewRequired.filter(pr => moment().diff(moment(pr.created_at), 'days') <= recentDays);
-  const lasting = reviewRequired.filter(pr => moment().diff(moment(pr.created_at), 'days') > recentDays);
+  const recent = [];
+  const lasting = [];
+  const readyForMerge = [];
+
+  for (const pr of prs) {
+    const reviews = await fetchReviewsForPR(pr.number);
+    const approvals = reviews.filter(review => review.state.toLowerCase() === 'approved');
+    const isReadyForMerge = approvals.length >= approvalsRequired;
```
[the remark this answers is not in the document]
Comment (universal-itengineer) [reply .github/scripts/fetch_prs.mjs:115]: I hope yes)

File: .github/scripts/fetch_prs.mjs (hunk 2 above)
Comment (diafour) [reply .github/scripts/fetch_prs.mjs:88]:
```

function formatReviewer(reviewer, details) {
  if (!details) {
    return reviewer.login;  
  }
  const loopName = details.name ? details.name.replace(/ /g, '.').toLowerCase() : 'Set name in profile!';
  return `${details.login} (@${loopName})`;
}

async function formatPR(pr) {
...
  const uniqueLogins = new Set();
  const fetchedReviewers = [];

  // Add requested reviewers
  if (pr.requested_reviewers && pr.requested_reviewers.length > 0) {
    for (const reviewer of pr.requested_reviewers) {
       uniqueLogins.add(details.login);
       const details = await fetchReviewerDetails(reviewer.login);
       fetchedReviewers.push(formatReviewer(reviewer, details));
    }
  }

  // Add more reviewers from reviews.
  const reviews = await fetchReviewsForPR(pr.number);
  for (const review of reviews) {
    if (uniqueLogins.has(review.user.login)) {
      continue;
    }
    uniqueLogins.add(review.user.login);
    const details = await fetchReviewerDetails(review.user.login);
    fetchedReviewers.push(formatReviewer(reviewer, details));
  }
```

Comment (universal-itengineer) [reply .github/scripts/fetch_prs.mjs]: Fixed

**Review by diafour — APPROVED** [review]

**Pull request merged**